### PR TITLE
[shopsys] fix error in javascript validation of ChoiceType

### PR DESCRIPTION
**Pull request by DavidOstrozlik**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Javascript validation throws error when you have ChoiceType with checkboxes
|New feature| No 
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| No 
|Fixes issues| #374  
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: packages/framework/src/Form/JsFormValidatorFactory.php [hunk 1]
```diff
@@ -62,7 +62,7 @@ protected function normalizeViewTransformers(FormInterface $form, array $viewTra
                 }
             }
 
-            array_push($transformer['choiceList'], $optionsItemsThatAreNotInstanceOfParameterValue);
+            $transformer['choiceList'] = array_merge($transformer['choiceList'], $optionsItemsThatAreNotInstanceOfParameterValue);
```
Comment (boris-brtan) [inline packages/framework/src/Form/JsFormValidatorFactory.php]: from the previous PR https://github.com/shopsys/shopsys/pull/89/files this looks like it was implemented intentionally.
is this OK ? should not make more sense to remove the whole functionality?
example:
```
$transformer['choiceList'] == [1=>ParameterValue, 3=>ParameterValue] 
$optionsItemsThatAreNotInstanceOfParameterValue == [0=>Brand,1=>Flag,2=>Brand,3=>Flag]
array_push == [1=>ParameterValue, 3=>ParameterValue] , 4 => [0=>Brand,1=>Flag,2=>Brand,3=>Flag]]
array_merge == [0=>ParameterValue, 1=>ParameterValue] , 2=>Brand, 3=>Flag, 4=>Brand, 5=>Flag]
```
is also this change connected with the ProductFilterFormType? Should not be that solved in another commit?

File: project-base/src/Shopsys/ShopBundle/Form/Front/Product/ProductFilterFormType.php [hunk 2]
```diff
@@ -54,7 +54,6 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'choices' => $config->getFlagChoices(),
                 'choice_label' => 'name',
                 'choice_value' => 'id',
-                'choice_name' => 'id',
```
[the remark this answers is not in the document]
Comment (boris-brtan) [reply project-base/src/Shopsys/ShopBundle/Form/Front/Product/ProductFilterFormType.php]: https://symfony.com/doc/current/reference/forms/types/choice.html#choice-name

[the remark this answers is not in the document]
Comment (DavidOstrozlik) [reply project-base/src/Shopsys/ShopBundle/Form/Front/Product/ProductFilterFormType.php]: I don't think so. The filtering works and JS doesn't throw any error

File: packages/framework/src/Form/JsFormValidatorFactory.php (hunk 1 above)
Comment (DavidOstrozlik) [reply packages/framework/src/Form/JsFormValidatorFactory.php]: This change is related to the problem so i think it should be in this commit
